### ISSUE TITLE
Header comment fixer regression

### ISSUE THREAD

**Issue by cs278** [FriendsOfPHP/PHP-CS-Fixer#4925]
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4870 seems to have broken `HeaderCommentFixer` in PHP-CS-Fixer ≥ 2.16.2. It is now removing other comments in the top scope.

When reporting an issue (bug) please provide the following information:

#### The PHP version you are using (`$ php -v`):
=> 7.4.4

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
=> 2.16.3

#### The command you use to run PHP CS Fixer:
=> vendor/bin/php-cs-fixer fix test.php

#### The configuration file you are using, if any:
```php
<?php

return PhpCsFixer\Config::create()
    ->setRules([
        'header_comment' => [
            'header' => 'header',
            'location' => 'after_open',
        ],
    ])
;
```

#### If applicable, please provide minimum samples of PHP code (as plain text, not screenshots):
 * before running PHP CS Fixer (no changes):
```php
<?php

/*
 * header
 */

declare(strict_types=1);

// Do not allow this script to be used on web
if ('cli' !== PHP_SAPI) {
    http_response_code(500);
    exit(1);
}
```

 * with unexpected changes applied when running PHP CS Fixer:
```php
<?php

/*
 * header
 */

declare(strict_types=1);
if ('cli' !== PHP_SAPI) {
    http_response_code(500);
    exit(1);
}
```

 * with the changes you expected instead:
```php
<?php

/*
 * header
 */

declare(strict_types=1);

// Do not allow this script to be used on web
if ('cli' !== PHP_SAPI) {
    http_response_code(500);
    exit(1);
}
```

**Comment by gharlan** [comment]
> #4870 seems to have broken `HeaderCommentFixer` in PHP-CS-Fixer ≥ 2.16.2.

I guess it is caused by #4895, not by #4870.

**Comment by julienfalque** [comment]
Should be fixed by #4936.

**Comment by SpacePossum** [comment]
fixed in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4936
